tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -2840,73 +2852,141 @@ NodeDef GetAddNNodeDef(const std::vector<string>& input_names, DataType dtype) {
   return op.operation.node()->def();
 }
 
-template <DataType dtype>
-void TestAddN(OpConverterTest* test) {
-  typedef typename EnumToDataType<dtype>::Type CType;
+struct AddNTestParams {
+  std::vector<float> input_value;
+  std::vector<string> input_names;
+  std::vector<int> expected_dim;
+  std::vector<float> expected_output;
+  Status status;
+};
+
+void TestAddN(ParameterizedOpConverterTestBase* test, AddNTestParams& p) {
   {
     // All inputs are tensors.
     test->Reset();
-    DataVec input_data;
-    for (const auto name : {"inp1", "inp2", "inp3"}) {
-      nvinfer1::DataType trt_type;
-      TF_ASSERT_OK(TfTypeToTrtType(dtype, &trt_type));
-      test->AddTestTensor(name, /*dims=*/{1, 2}, /*batch_size=*/2, trt_type);
-      input_data.push_back({name, test->AsTensor<CType>({CType(1), CType(2),
-                                                         CType(3), CType(4)})});
-    }
-    const NodeDef node_def = GetAddNNodeDef({"inp1", "inp2", "inp3"}, dtype);
-    test->RunValidationAndConversion(node_def);
-
-    TRT_TensorOrWeights output;
-    TF_EXPECT_OK(test->GetTensorOrWeights("my_addn", &output));
-    ASSERT_TRUE(output.is_tensor());
-    ExpectTrtDimsEqualsArray({1, 2}, output.tensor()->getDimensions());
+    const NodeDef node_def = GetAddNNodeDef(p.input_names, test->get_tf_type());
 
-    DataVec output_data{{"my_addn", test->ConstructTensor<CType>(4)}};
-    TF_EXPECT_OK(test->BuildAndRun(input_data, &output_data, /*batch_size=*/2));
-    EXPECT_THAT(GetSpanForData<CType>(output_data[0]),
-                ElementsAreArray(CastTestVector<int, CType>({3, 6, 9, 12})));
-  }
-  {
-    // Input contains tensors and weights.
-    test->Reset();
     DataVec input_data;
-    for (const auto name : {"inp1", "inp2"}) {
-      nvinfer1::DataType trt_type;
-      TF_ASSERT_OK(TfTypeToTrtType(dtype, &trt_type));
-      test->AddTestTensor(name, /*dims=*/{1, 2}, /*batch_size=*/1, trt_type);
-      input_data.push_back({name, test->AsTensor<CType>({CType(1), CType(2)})});
+    int input_offset = 0;
+    const int window_size =  p.input_value.size() / p.input_names.size();
```

Comment:
Shall we check that p.input_value.size() is a multipler of p.input_names.size()?